internal/authorization/authorization.go

```diff
@@ -0,0 +1,40 @@
+package authorization
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+)
+
+const (
+	DriverNoop    string = "noop"
+	DriverOpenFGA string = "openfga"
+)
+
+var authorizers = map[string]func() Authorizer{
+	DriverNoop:    func() Authorizer { return &NoopAuthorizer{} },
+	DriverOpenFGA: func() Authorizer { return &OpenFGAAuthorizer{} },
+}
+
+type Authorizer interface {
+	FilterObjects(context.Context, string, string, string, []string) ([]string, error)
+	Check(context.Context, string, string, string) (bool, error)
+	init(string, string, string, string, string, bool, logging.LoggerInterface) error
+}
+
+// Init the authorizer based on the config
+func NewAuthorizer(apiScheme, apiHost, storeID, apiToken, authModelID string, debug bool, logger logging.LoggerInterface) (*Authorizer, error) {
```

Comment:
so, sticky point here, i'd separate the low level client with a NewClient like we do in the other internals, and then have a NewAuthorizer that accepts a client pointer, based on that it's a Noop or not, and there u ahve all the high level operations u need